ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -94,7 +96,7 @@ public Driver(HiveConf conf) {
 
   // Pass lineageState when a driver instantiates another Driver to run or compile another query
   public Driver(HiveConf conf, Context ctx, LineageState lineageState) {
-    this(QueryState.getNewQueryState(conf, lineageState), null);
+    this(QueryState.getNewQueryState(conf, lineageState), QueryInfo.getFromConf(conf));
```

Comment:
why is this change needed?

Reply:
in the beginning of the implementation, I ran full precommit tests with enabling this service by default and enabling everywhere it's possible (all qtests for example), which led to some problems that are not visible otherwise
one those problems were that the Driver instance can be created here with null QueryInfo, which led to NPEs in both prod and test code (as far as I can remember), so I decided to create an empty one here, which looks much more valid then null (so actually, the query history service was the one that showed QueryInfo==null can cause problems)